### PR TITLE
Adds throughput support for Cosmos DB Mongo Collection

### LINKED ISSUE #3586: Allow to specify throughput for azurerm_cosmosdb_mongo_collection
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We need to specify the throughput for a collection. As this effects the performance and the costs this feature is really needed to use this resource.

### New or Affected Resource(s)

azurerm_cosmosdb_mongo_collection

### Potential Terraform Configuration

```hcl
resource "azurerm_cosmosdb_mongo_database" "example" {
  name                = "tfex-cosmos-mongo-db"
  resource_group_name = "${data.azurerm_cosmosdb_account.example.resource_group_name}"
  account_name        = "${data.azurerm_cosmosdb_account.example.name}"
  throughput = "400"
}
```

### References
Azure CLI
https://docs.microsoft.com/en-us/cli/azure/cosmosdb/collection?view=azure-cli-latest#az-cosmosdb-collection-create

### PR DESCRIPTION
**Pull request by kmfk**
This would resolve #3586 and allow setting the `throughput` property on the collection.

There are several places where support should be added for `throughput` on the Cosmos DB resources. If this ok, I can add in the others.

### CODE REVIEW

File: website/docs/r/cosmosdb_mongo_collection.html.markdown [hunk 1]
```diff
@@ -53,7 +54,8 @@ The following arguments are supported:
 * `resource_group_name` - (Required) The name of the resource group in which the Cosmos DB Mongo Collection is created. Changing this forces a new resource to be created.
 * `database_name` - (Required) The name of the Cosmos DB Mongo Database in which the Cosmos DB Mongo Collection is created. Changing this forces a new resource to be created.
 * `default_ttl_seconds` - (Required) The default Time To Live in seconds. If the value is `-1` items are not automatically expired.
-* `shard_key` - (Required) The name of the key to partition on for sharding. There must not be any other unique index keys. 
+* `shard_key` - (Required) The name of the key to partition on for sharding. There must not be any other unique index keys.
+* `throughput` - (Optional) The throughput of MongoDB collection (RU/s). Must be set in increments of 100. Default value is 400.
```
Comment (aequasi) [inline website/docs/r/cosmosdb_mongo_collection.html.markdown]:
```suggestion
* `throughput` - (Optional) The throughput of the MongoDB collection (RU/s). Must be set in increments of 100. The default value is 400.
```

Comment (katbyte) [reply website/docs/r/cosmosdb_mongo_collection.html.markdown]: If 400 is the minimum we should state that here to

File: azurerm/resource_arm_cosmosdb_mongo_collection.go [hunk 2]
```diff
@@ -66,6 +66,13 @@ func resourceArmCosmosDbMongoCollection() *schema.Resource {
 				ValidateFunc: validation.IntAtLeast(-1),
 			},
 
+			"throughput": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      400,
+				ValidateFunc: validation.IntAtLeast(400),
```
Comment (katbyte) [inline azurerm/resource_arm_cosmosdb_mongo_collection.go]: We should validate that it is increments of 100 & is at least 400

File: website/docs/r/cosmosdb_mongo_collection.html.markdown (hunk 1 above)
Comment (kmfk) [reply website/docs/r/cosmosdb_mongo_collection.html.markdown]: done

File: azurerm/resource_arm_cosmosdb_mongo_collection.go (hunk 2 above)
Comment (kmfk) [reply azurerm/resource_arm_cosmosdb_mongo_collection.go]: I added a validator that checks for a minimum of 400 and that its set in increments of 100 - I believe this logic applies to other CosmosDB resources as well, so I added it to the existing `cosmos` validators.

I also saw that there were no test cases for the existing cosmos validators, so I added them in

**Comment by kmfk** [comment]
@katbyte I believe this is good. 

Pretty much all CosmosDB resources can use the `throughput` parameter, but I believe support for it was added to the go sdk after these CosmosDB resources were added to this provider. 

I don't mind adding support for the rest of the resources - do you prefer a separate PR for those?

**Comment by ragoragino** [comment]
Hii @kmfk,

just wanted to ask if you plan to open a PR to add support for defining throughput also for other resources? We would be excited to see it in the case of database creation.

Thanks.

**Comment by kmfk** [comment]
@ragoragino Yeah, I'll take a stab at the others

**Pull request merged**